### PR TITLE
formatting decision for pipes

### PR DESCRIPTION
**Pull request by awalterschulze**

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
❤️

File: doc/FormattingDecisionPipes.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+grep -r -E --include "*\.hrl" --include "*\.erl" '\w+\s*\|\s*$' . | grep -v -E '\|.*\|' | grep -v -E '(\]|\[|%)' | wc -l
+# count lines that start with a pipe (prefix)
+grep -r -E --include "*\.hrl" --include "*\.erl" '^\s+\|\s*\w+' . | grep -v -E '\|.*\|' | grep -v -E '(\]|\[|%)' | wc -l
+```
+
+We found that both OTP and WhatsApp prefer pipes as a prefix.
+
+OTP:
+  - Pipes as a Suffix: 531
+  - Pipes as a Prefix: 734
```
Comment (michalmuskala) [inline doc/FormattingDecisionPipes.md:47]: These numbers seem suspiciously low to me for around 2 MLoC of source code. I would expect on the order of tens of thousands

Comment (awalterschulze) [reply doc/FormattingDecisionPipes.md:47]: Fair enough, where do you think the regex can be improved?

Comment (paulo-ferraz-oliveira) [reply doc/FormattingDecisionPipes.md:47]: Oh, noooo...

Comment (michalmuskala) [reply doc/FormattingDecisionPipes.md:47]: I'm actually not sure.. Regexes are too confusing. Let's go with those ones

**Review by michalmuskala — APPROVED** [review]

**Pull request merged**